jerry-core/parser/js/js-lexer.c

```diff
@@ -304,42 +304,51 @@ typedef struct
   lexer_token_type_t type;  /**< keyword token type */
 } keyword_string_t;
 
+/**
+ * Keyword list data.
+ */
+typedef struct
+{
+  const keyword_string_t *keyword_p; /**< keyword string list */
+  const uint8_t length;  /**< keyword list length */
+} keyword_string_list_t;
+
 /**
  * @{
  * Keyword defines
  */
 #define LEXER_KEYWORD(name, type) { (const uint8_t *) (name), (type) }
-#define LEXER_KEYWORD_END()       { (const uint8_t *) NULL, LEXER_EOS }
+#define LEXER_KEYWORD_LIST(name) \
+  { (const keyword_string_t *) (name), (const uint8_t) (sizeof ((name)) / sizeof ((name)[0])) }
+#define LEXER_KEYWORD_END()
 /** @} */
 
 /**
  * Keywords with 2 characters.
  */
-static const keyword_string_t keyword_length_2[4] =
+static const keyword_string_t keyword_length_2[] =
 {
   LEXER_KEYWORD ("do", LEXER_KEYW_DO),
   LEXER_KEYWORD ("if", LEXER_KEYW_IF),
   LEXER_KEYWORD ("in", LEXER_KEYW_IN),
-  LEXER_KEYWORD_END ()
 };
 
 /**
  * Keywords with 3 characters.
  */
-static const keyword_string_t keyword_length_3[6] =
+static const keyword_string_t keyword_length_3[] =
```

Comment:
This is just an idea, but what about renaming these to `keywords_with_length_3` or `keywords_of_length_3`, I don't know which one is correct. Since we touch this code anyway. Or `keyword_group_with_length_3`. I am open to other suggestions.